### PR TITLE
fix: flaky test `Request-queue UI changes should gracefully handle deleted network @no-mmi.`

### LINKED ISSUE #27304: Fix "Request-queue UI changes should gracefully handle de..." flaky tests
CI failure link reported on Sept 24
https://app.circleci.com/pipelines/github/MetaMask/metamask-extension/101984/workflows/4668d4c0-22db-4750-b0b9-3b95f5661ac0/jobs/3798329/tests

https://app.circleci.com/pipelines/github/MetaMask/metamask-extension/101369/workflows/6c8551cb-1b8c-4ff5-a912-889d65e289b2/jobs/3774008/tests

Build - Regular
Browser - Chrome
Test Scenario - Request-queue UI changes should gracefully handle deleted network @no-mmi
Test Path - /test/e2e/tests/request-queuing/ui.spec.js

```
TypeError: Cannot destructure property 'metamask' of 'notificationWindowState' as it is null.
  (Ran on CircleCI Node 0 of 20, Job test-e2e-chrome)
    at switchToDialogPopoverValidateDetails (test/e2e/tests/request-queuing/ui.spec.js:111:15)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async /home/circleci/project/test/e2e/tests/request-queuing/ui.spec.js:434:9
    at async withFixtures (test/e2e/helpers.js:217:5)
    at async Context.<anonymous> (test/e2e/tests/request-queuing/ui.spec.js:369:5)
```

### PR DESCRIPTION
**Pull request by seaona**
## **Description**

The problem is that we are trying to get state:

`window.stateHooks?.getCleanAppState?.(),`

 before it is fully loaded, making the `metamask` property being undefined

```
TypeError: Cannot destructure property 'metamask' of 'notificationWindowState' as it is null.
  (Ran on CircleCI Node 0 of 20, Job test-e2e-chrome)
    at switchToDialogPopoverValidateDetails (test/e2e/tests/request-queuing/ui.spec.js:111:15)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async /home/circleci/project/test/e2e/tests/request-queuing/ui.spec.js:434:9
    at async withFixtures (test/e2e/helpers.js:217:5)
    at async Context.<anonymous> (test/e2e/tests/request-queuing/ui.spec.js:369:5)
```

We could add a retry logic, but I found that if we make sure controllers are loaded, then test passes 100%, so I think there's no need for retry logic.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/27393?quickstart=1)

## **Related issues**

Fixes: [Request-queue UI changes should gracefully handle de...](https://github.com/MetaMask/metamask-extension/issues/27304)

## **Manual testing steps**

1. Check ci
2. Run test locally `yarn test:e2e:single test/e2e/tests/request-queuing/ui.spec.js --browser=chrome --leave-running=true`

## **Screenshots/Recordings**

Screenshot from 2024-09-25 15-59-06

See how you some UI is loaded before the controller css class is added / controllers are ready and state is ready.

https://github.com/user-attachments/assets/5f8a9777-d09a-439c-b6c6-4afd68094189

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: test/e2e/webdriver/driver.js [hunk 1]
```diff
@@ -777,14 +777,27 @@ class Driver {
     const response = await this.driver.get(`${this.extensionUrl}/${page}.html`);
     // Wait for asynchronous JavaScript to load
     if (waitForControllers) {
-      await this.driver.wait(
-        until.elementLocated(this.buildLocator('.controller-loaded')),
-        10 * 1000,
-      );
```
Comment (seaona) [inline test/e2e/webdriver/driver.js:783]: just abstracted this into a function, so we can use re-use it in other places (like the ui test)

**Review by bergeron — DISMISSED** [review]

File: test/e2e/webdriver/driver.js [hunk 2]
```diff
@@ -777,14 +777,27 @@ class Driver {
...
+      this.waitForControllersLoaded();
```
Comment (darkwing) [inline test/e2e/webdriver/driver.js:780]: Does this need to be `await`'d?

Comment (seaona) [reply test/e2e/webdriver/driver.js:780]: yeah!! thank you @darkwing

Comment (seaona) [inline test/e2e/webdriver/driver.js:780]:
```suggestion
      await this.waitForControllersLoaded();
```

**Review by DDDDDanica — APPROVED** [review]

**Review by bergeron — APPROVED** [review]

**Review by hjetpoluru — APPROVED** [review]
LGTM

**Review by darkwing — APPROVED** [review]

**Pull request merged**